### PR TITLE
Tweak config for mode=max-autotune/reduce-overhead

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94659

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/__init__.py [hunk 1]
```diff
@@ -1335,15 +1335,19 @@ def __init__(self, mode, options, dynamic):
...
-        if mode is None:
-            return
-        elif mode == "default":
+        if mode is None or mode == "default":
             pass
         elif mode == "reduce-overhead":
-            self.config["triton.cudagraphs"] = True
+            self.apply_options({
+                "triton.cudagraphs": True,
+                "size_asserts": False,
```
Comment (Chillee) [inline torch/__init__.py:1343]: I think if we're using cudagraphs size_asserts wouldn't have any affect anyways?

File: torch/_inductor/config.py [hunk 2]
```diff
@@ -157,10 +154,6 @@ class triton:
     convolution = "aten"
 
     # Always load full blocks (rather than broadcasting inside the block)
-    # Set default as True because otherwise will encouter `map::at` error
```
Comment (Chillee) [inline torch/_inductor/config.py:160]: Is this still an issue?

**Review by Chillee — APPROVED** [review]

File: torch/__init__.py (hunk 1 above)
Comment (jansel) [reply torch/__init__.py:1343]: It helps if cudagraph gets disabled (due to mutation, etc) and on CPU.

File: torch/_inductor/config.py (hunk 2 above)
Comment (jansel) [reply torch/_inductor/config.py:160]: It has been fixed.

Comment (Chillee) [reply torch/_inductor/config.py:160]: Then can we just remove this option?

Comment (jansel) [reply torch/_inductor/config.py:160]: Yeah possibly. It still might be useful for debugging Triton issues.

**Pull request closed without merging**